File: Books/Books/Math_Refresher/python/maxima_minima.py
# maxima_minima.py
# Maxima and Minima
#-------------------------------------------------------------------------
import numpy as np
import matplotlib.pyplot as plt

plt.close('all')

# f(x) = x^2 + 2 and its derivative f'(x) = 2x
x = np.arange(-4, 4.1, 0.1)
fx = np.zeros((len(x),))
dx = np.zeros((len(x),))
for ii in range(len(x)):
    fx[ii] = x[ii]**2 + 2
    dx[ii] = 2 * x[ii]

plt.figure(figsize=(10, 5))

plt.subplot(1, 2, 1)
plt.plot(x, fx, '.', linewidth=2)
plt.plot(x, fx, 'b')
plt.xlabel(r'$$x$$', fontsize=12)
plt.ylabel(r'$$f(x)$$', fontsize=12)
plt.title(r'$$f(x) = x^2 + 2$$', fontsize=12)

plt.subplot(1, 2, 2)
plt.plot(x, dx, '.', linewidth=2)
plt.plot(x, dx, 'b')
plt.xlabel(r'$$x$$', fontsize=12)
plt.ylabel(r"$$f'(x)$$", fontsize=12)
plt.title(r"$$f'(x) = 2x$$", fontsize=12)

plt.savefig("../figures/maxima_minima.pdf")
plt.show()

File: Books/Books/Math_Refresher/matlab/maxima_minima.m
% Maxima and Minima

% Start of script
%-------------------------------------------------------------------------%
close all;                   	% close all figures
clearvars; clearvars -global;	% clear all variables
clc;                         	% clear the command terminal
format shortG;                 	% pick the most compact numeric display
format compact;                	% suppress excess blank lines

%f(x) = x^2 + 2
%dx(x) = 2x
x = -4:0.1:4;
fx = zeros(length(x),1);
dx = zeros(length(x),1);
for ii = 1:length(x)
    fx(ii,1) = x(ii)^2  + 2;
    dx(ii,1) = 2*x(ii);
end


figure(1)
subplot(1,2,1)
    plot(x,fx,'.','LineWidth',2)
    hold on
    plot(x,fx,'b')
    xlabel('$$x$$','Interpreter','latex')
    ylabel('$$f(x)$$','Interpreter','latex')
    title('$$f(x) = x^2 + 2$$','Interpreter','latex')
subplot(1,2,2)
    plot(x,dx,'.','LineWidth',2)
    hold on
    plot(x,dx,'b')
    xlabel('$$x$$','Interpreter','latex')
    ylabel('$$f''(x)$$','Interpreter','latex')
    title('$$f''(x) = 2x$$','Interpreter','latex')
    
save_all_figs_OPTION('../figures/maxima_minima','pdf')